STYLING.md

```diff
@@ -0,0 +1,225 @@
+# React Native Styling Guidelines
+
+## Where to Define Styles
+
+All styles must be defined in the `styles.js` file which exists as a globally exported stylesheet of sorts for the application. Unlike some React Native applications we are not using `StyleSheet.create()` and instead store styles as plain JS objects. There are many helper styles available for direct use in components and can be found in the `/styles` directory.
```

Comment:
I've also used a pattern a few times of creating functions in a style file to create dynamic styles like [here](https://github.com/Expensify/Expensify.cash/blob/1017f44cd0701a0c00da122970ec2aa29c5cc64c/src/styles/getReportActionItemStyles.js) and [here](https://github.com/Expensify/Expensify.cash/blob/7a2d5a5c3b4f64e3b051310c1b3ccd05c7e3e007/src/styles/getModalStyles.js). What are our thoughts on that? I sort of think it's fine, but we could also move those same functions into `style.js`.

Reply:
Some similar functions already live in `styles.js` [here](https://github.com/Expensify/Expensify.cash/blob/d81e645090b213b4d5758d507af670b12703f5b8/src/styles/styles.js#L1352-L1405), so we haven't been consistent there.

I really don't have a preference whether those dynamic style generators live in their own files or `styles.js`, but we should be consistent.

Reply:
Yeah I think those function files can be modular but maybe we enforce that they are imported via `styles.js` directly so they are all documented there at least?

Reply:
I like the idea of modularizing those methods better. I think that `style.js` should just be in charge of collecting and being the final export of our styles. There are a couple minor changes I'd like to request for this paragraph.

1. All styles must be defined in the `/styles` directory and `style.js` contains the final export after gathering all appropriate styles.
1. Remove "of sorts"